zookeeper-server/src/main/java/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -1003,6 +1004,32 @@ public static List<ACL> fixupACL(String path, List<Id> authInfo, List<ACL> acls)
             throw new KeeperException.InvalidACLException(path);
         }
         List<ACL> rv = new ArrayList<>();
+
+        // Overwrite the acl list for users (except super user) when znode group acl feature is on
+        // Set znode ACL to the corresponding znode group acl designated ACL instead
+        // For single domain user / cross domain components -> (x509 : domainName)
+        // For user w/o a domain -> (x509: clientURI)
+        if (X509AuthenticationConfig.getInstance().isX509ClientIdAsAclEnabled() && ProviderRegistry
+            .getServerProvider(
+                X509AuthenticationUtil.X509_SCHEME) instanceof X509ZNodeGroupAclProvider
+            && !X509AuthenticationConfig.getInstance().isDedicatedServerEnabled()) {
+            authInfo.forEach(id -> {
+                if (id.getScheme().equals(X509AuthenticationUtil.X509_SCHEME)) {
+                    overwriteAclListToCreatorId(uniqacls, id);
+                } else if (id.getScheme().equals(X509AuthenticationUtil.SUPERUSER_AUTH_SCHEME)
```

Comment:
It seems both branches are doing the same thing. In that case, can't they be merged?

Reply:
I think this kind of stuff would get optimized away by the compiler anyway, so I think if separating the `if`s make it more readable and clearly illustrate that they are covering different scenarios, I think it's ok to leave them separate

Reply:
I'm open to keeping it as is (with the addition of comments as to why it is the way it is), but if readability is the goal then two separate booleans can be defined with comments explaining what each checks, and the `if` condition can be the `OR` of both. Example:
```
boolean isX509Scheme = id.getScheme().equals(X509AuthenticationUtil.X509_SCHEME);
boolean isCrossDomain = id.getScheme().equals(X509AuthenticationUtil.SUPERUSER_AUTH_SCHEME) &&
              !(id.getId()).equals(X509AuthenticationConfig.getInstance().getZnodeGroupAclSuperUserId());
if (isX509Scheme || isCrossDomain) {
...
}
```

Reply:
See above.

Reply:
> Reference
I adopted this suggestion and updated the code. Thanks